File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/observable/g1_java/BankAccount.java
package com.twsg.lnl.kotlin.c5_delegate.observable.g1_java;

import java.util.function.BiConsumer;

/**
 * @author tonyl
 */
public class BankAccount {

    private final Observable<Double> balance;

    public BankAccount(double balance, BiConsumer<Double, Double> balanceObserver) {
        this.balance = new Observable<>(balance, balanceObserver);
    }

    public double getBalance() {
        return this.balance.get();
    }

    public void setBalance(double balance) {
        this.balance.set(balance);
    }
}

File: src/test/java/com/twsg/lnl/kotlin/c5_delegate/observable/g2_kotlin/BankAccount.kt
package com.twsg.lnl.kotlin.c5_delegate.observable.g2_kotlin

import kotlin.properties.Delegates.observable

/**
 * @author tonyl
 */

class BankAccount(initialBalance: Double, observer: (Double, Double) -> Unit) {

    var balance: Double by observable(initialBalance) { _, old, new ->
        observer(old, new)
    }

}